test/response_test.rb

```diff
@@ -852,9 +852,237 @@ class RubySamlTest < Minitest::Test
+      settings.issuer = 'http://rubysaml.com:3000/saml/metadata'
+      settings.assertion_consumer_service_url = 'http://rubysaml.com:3000/saml/acs'
+      settings.certificate = ruby_saml_cert_text
+      settings.private_key = ruby_saml_key_text
+    end
+
+    it 'able when signed_message_encrypted_unsigned_assertion' do
+      response = OneLogin::RubySaml::Response.new(signed_message_encrypted_unsigned_assertion, :settings => settings)
+      Timecop.freeze(Time.parse("2015-03-19T14:30:31Z")) do
+        assert response.is_valid?
+        assert_empty response.errors
+        assert_equal "test", response.attributes[:uid]
+        assert_equal "98e2bb61075e951b37d6b3be6954a54b340d86c7", response.name_id
+      end
+    end
+
+    it 'able when signed_message_encrypted_signed_assertion' do
+      response = OneLogin::RubySaml::Response.new(signed_message_encrypted_signed_assertion, :settings => settings)
+      Timecop.freeze(Time.parse("2015-03-19T14:30:31Z")) do
+        assert response.is_valid?
+        assert_empty response.errors
+        assert_equal "test", response.attributes[:uid]
+        assert_equal "98e2bb61075e951b37d6b3be6954a54b340d86c7", response.name_id
+      end
+    end
+
+    it 'able when unsigned_message_encrypted_signed_assertion' do
+      response = OneLogin::RubySaml::Response.new(unsigned_message_encrypted_signed_assertion, :settings => settings)
+      Timecop.freeze(Time.parse("2015-03-19T14:30:31Z")) do
+        assert response.is_valid?
+        assert_empty response.errors
+        assert_equal "test", response.attributes[:uid]
+        assert_equal "98e2bb61075e951b37d6b3be6954a54b340d86c7", response.name_id
+      end
+    end
+
+    it 'unable when unsigned_message_encrypted_unsigned_assertion' do
+      response = OneLogin::RubySaml::Response.new(unsigned_message_encrypted_unsigned_assertion, :settings => settings)
+      Timecop.freeze(Time.parse("2015-03-19T14:30:31Z")) do
+        assert !response.is_valid?
+        assert_includes response.errors, "Found an unexpected number of Signature Element. SAML Response rejected"
+      end
+    end
+  end
+
+  describe "#decrypt_assertion" do
+    before do
+      settings.private_key = ruby_saml_key_text
+    end
+
+    describe "check right settings" do
+
+      it "unable to decrypt the assertion if no private key" do
+        response = OneLogin::RubySaml::Response.new(signed_message_encrypted_unsigned_assertion, :settings => settings)        
+
+        encrypted_assertion_node = REXML::XPath.first(
+          response.document,
+          "(/p:Response/EncryptedAssertion/)|(/p:Response/a:EncryptedAssertion/)",
+          { "p" => "urn:oasis:names:tc:SAML:2.0:protocol", "a" => "urn:oasis:names:tc:SAML:2.0:assertion" }
+        )
+        response.settings.private_key = nil
+
+        error_msg = "An EncryptedAssertion found and no SP private key found on the settings to decrypt it"
```

Comment:
It should not change, if change for a reason, we may update the tests related.
Provide standard errors is something that help developers ;)